slack-teamspeak.py

```diff
@@ -29,13 +29,18 @@ def update_users(last_users):
     clientlist = srv.clientlist()
     current_users = {client['client_database_id'] for client in clientlist.values() if client['client_type'] == '0'}
     new_users = current_users - last_users
-    return sorted(client['client_nickname'] for client in clientlist.values() if client['client_database_id'] in new_users), current_users
+	former_users = sorted(client['client_nickname'] for client in last_users if client not in current_users)
+    return sorted(client['client_nickname'] for client in clientlist.values() if client['client_database_id'] in new_users), current_users, former_users
```

Comment:
Unless I'm missing something, this doesn't work because you're only returning the nicknames in `current_users` but are assuming it contains the entire `clientlist.values()` object for line 32.